PortalSwift/Sources/PortalSwift/Core/Api/PortalApi.swift

```
@@ -0,0 +1,589 @@
+      } catch {
+        self.logger.error("PortalApi.getBalances() - Unable to eject: \(error.localizedDescription)")
+        throw error
+      }
+    }
+
+    self.logger.error("PortalApi.eject() - Unable to build request URL.")
+    throw URLError(.badURL)
+  }
+
+  public func getBalances(_ chainId: String) async throws -> [FetchedBalance] {
+    if let url = URL(string: "\(baseUrl)/api/v3/clients/me/balances?chainId=\(chainId)") {
+      do {
+        let data = try await get(url, withBearerToken: self.apiKey)
+        let balancesResponse = try decoder.decode([FetchedBalance].self, from: data)
+
+        return balancesResponse
+      } catch {
+        self.logger.error("PortalApi.getBalances() - Unable to get balanaces: \(error.localizedDescription)")
+        throw error
+      }
+    }
+
+    self.logger.error("PortalApi.getBalances() - Unable to build request URL.")
+    throw URLError(.badURL)
+  }
+
+  /// Retrieve the client by API key.
+  /// - Returns: ClientResponse
+  public func getClient() async throws -> ClientResponse {
+    if let url = URL(string: "\(baseUrl)/api/v3/clients/me") {
+      do {
+        let data = try await get(url, withBearerToken: self.apiKey)
+        let clientResponse = try decoder.decode(ClientResponse.self, from: data)
+
+        return clientResponse
+      } catch {
+        throw error
+      }
+    }
+
+    throw URLError(.badURL)
+  }
+
+  public func getQuote(_ swapsApiKey: String, withArgs: QuoteArgs) async throws -> Quote {
+    if let url = URL(string: "\(baseUrl)/api/v1/swaps/quote") {
+      // Build the request body
+      var body = withArgs.toDictionary()
+      // Append Portal-provided values
+      body["address"] = AnyEncodable(self.address)
+      body["apiKey"] = AnyEncodable(swapsApiKey)
+      body["chainId"] = AnyEncodable(self.chainId)
+
+      let data = try await post(url, withBearerToken: self.apiKey, andPayload: body)
+      let response = try decoder.decode(Quote.self, from: data)
```

Should we wrap this with a do catch too in order to log?

---

We actually want these to throw if something fails. That's one of the fundamental changes to our approach in the SDK: stop swallowing errors 😅